ranger: 1.9.3 -> 1.9.3-unstable-2023-08-23

9ary opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

ranger hasn't seen a new release for years, but development has been continuing, and some interesting new features (such as sixel image previews) are only available in master.

I don't expect this patch to be merged, but it's still useful, so I'm submitting it as a draft.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

9ary committed:
ranger: 1.9.3 -> unstable-2023-08-23

donovanglover commented:
+1. For ranger alternatives I've been enjoying [joshuto](https://github.com/kamiyaa/joshuto) and [yazi](https://github.com/sxyazi/yazi) is cool as well.

magnetophon commented:
Thanks for the update!

I've switched to yazi; it has the features I need plus the speed I like!

toonn reviewed (approved):
As a Ranger maintainer I think this is fair, we're too slow in doing releases currently.

9ary commented:
Cool, I can rebase this and set it to non-draft then. Is there a particular commit that would make sense to pin, or just whatever is the latest? I see just one was added since I made this PR so we're not really chasing a moving target at the moment.

toonn commented:
I do believe tracking the latest is what we want. We tend to take care not to break the master branch, or fix it promptly when we do.

9ary commented:
Sounds good to me. Rebased, updated to point to the latest commit, and added a changelog entry.

magnetophon reviewed (approved):
Sorry for the late reply,
I didn't notice the PR got changed from draft to regular pr.

Result of `nixpkgs-review pr 280143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ranger</li>
    <li>ranger.dist</li>
  </ul>
</details>

Aleksanaa commented on pkgs/applications/file-managers/ranger/default.nix:
```diff
@@ -1,28 +1,30 @@
 python3Packages.buildPythonApplication rec {
   pname = "ranger";
-  version = "1.9.3";
+  version = "unstable-2023-08-23";
```
```suggestion
  version = "1.9.3-unstable-2023-08-23";
```

https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming:

> If a package is a commit from a repository without a version assigned, then the version attribute should be the latest upstream version preceding that commit, followed by -unstable- and the date of the (fetched) commit. The date must be in "YYYY-MM-DD" format.

> Example: Given a project had its latest releases 2.2 in November 2021, and 3.0 in January 2022, a commit authored on March 15, 2022 for an upcoming bugfix release 2.2.1 would have version = "2.2-unstable-2022-03-15".

toonn replied:
@9ary, can you incorporate this suggestion? (Add `Co-authored-by:` if you don't commit it via the GitHub UI.)

Aleksanaa replied:
I'll do it later

9ary replied:
Sorry, I forgot about this.

I wasn't aware of that convention (I don't think I've actually seen it in use before). Fixing this now.

9ary committed:
ranger: follow convention for unstable versions

If a package is a commit from a repository without a version assigned,
then the version attribute should be the latest upstream version
preceding that commit, followed by -unstable- and the date of the
(fetched) commit. The date must be in "YYYY-MM-DD" format.

Co-authored-by: aleksana <me@aleksana.moe>

toonn commented on nixos/doc/manual/release-notes/rl-2405.section.md:
```diff
@@ -353,6 +353,9 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 - New options were added to the dnsdist module to enable and configure a DNSCrypt endpoint (see `services.dnsdist.dnscrypt.enable`, etc.).
   The module can generate the DNSCrypt provider key pair, certificates and also performs their rotation automatically with no downtime.
 
+- The ranger file manager was updated from the latest release (which is a few
```
Should we even add a changelog for this? Package bumps without consequences to config or state files don't seem worth mentioning. What's the official policy on this?

9ary replied:
I've seen this done for package that may contain breaking changes. I'd say switching from a tagged release to tracking master is potentially a breaking change, but I'm neutral about this.

toonn replied:
As Ranger maintainer, I disagree. We pay a lot of attention to backwards compatibility and there haven't been big changes.

9ary replied:
It's less about compatibility and more about how close master is to production-quality. 100% fair though, I haven't really run into any regressions myself over a few months, so I'm fine with dropping the changelog entry.

Aleksanaa commented:
Please squash and reword (the version) your commits, other wise I will squash merge it

9ary commented:
No time to do this right now, feel free to squash merge. I made it a separate commit since @toonn asked me to add a co-authored-by trailer, but that's not relevant to the rest of the changes here.

Aleksanaa committed:
ranger: remove version test

Aleksanaa committed:
ranger: remove update note
